src/LoadOnline.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace AntoineFr\Online;
+
+use Carbon\Carbon;
+use Flarum\Api\Controller\ShowForumController;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Flarum\User\User;
+use Psr\Http\Message\ServerRequestInterface;
+use Tobscure\JsonApi\Document;
+
+class LoadOnline
+{
+    protected $settings;
+
+    public function __construct(SettingsRepositoryInterface $settings)
+    {
+        $this->settings = $settings;
+    }
+
+    public function __invoke(ShowForumController $controller, &$data, ServerRequestInterface $request, Document $document)
+    {
+        $actor = $request->getAttribute('actor');
+        $ago = Carbon::now()->subtract(5, 'minutes');
+
+        $online = User::query()->whereRaw('? < last_seen_at', [$ago])->get();
+
+        $filtered = collect($online)->map(function ($user) use ($actor) {
+            if ($user->getPreference('discloseOnline') || $actor->can('viewLastSeenAt', $user)) {
+                return $user;
+            }
+        });
+
+        $max = (int) $this->settings->get('antoinefr-online.displaymax', 5);
+
+        $data['online'] = $filtered->slice(0, $max <= 0 ? null : $max);
```

Comment:
I found a `take` method which could be used here

```php
        $data['online'] = $filtered->take($max);
```

Reply:
the reason i didn't used this is because if a user save the settings without input something, 0 is returned instead of 5. It looks like 5 is returned only if is null or undefined. or maybe you want this behavior?

Reply:
I think something like this would be nice: 

- max = -1 (or maybe < 0) -> display all online users
- max = 0 -> display no online user (could be used with `{more}` to only display `and XXX user(s)` without showing the usernames of online users)
- max > 0 -> display `max` users

What do you think?

Reply:
The second one you mean like `There are {more} users online` or `{more} users are online` ? i like it

Reply:
Yes, something like that. It is related to the first point you highlighted in this message: https://github.com/AntoineFr/flarum-ext-online/pull/26#issuecomment-835531948

> the `and {more} more...` is no longer used.

If `$max` is 2, we would have:
```
# Who is online?
(avatar) username
(avatar) username
and 5 more...
```

If `$max` is 0, we would have:
```
# Who is online?
and 7 more...
```

Of course it would need a text that works in both versions, or an `if` statement.